### ISSUE TITLE
How would I make this plugin work better with a forum system like phpbb? 

### ISSUE THREAD

**Issue by maxpen** [samclarke/SCEditor#437]
I've managed to integrate your plugin to a phpbb board but the quote feature and other bbcodes needs to be adjusted to work flawlessly or work at all. 

I can tell you already though they plan to add your plugin next version:
https://tracker.phpbb.com/browse/PHPBB3-10418

**Which may take some months but in the meantime what can I do to make it work good?**

I guess making a custom plugin? My js is rather limited. XD

**Also how do I define the path for the languages and emotions folder?**
I do not want to upload them in root but instead in my theme folder.

**Comment by brunoais** [comment]
**Hi!** I'm the one who is doing the integration of SCE<->phpBB. I already know those two fairly well so I think I can help you in what you need in this theme.
>Which may take some months [...]

It will come out in 3.2 if they don't complain of anything I'm doing.

> [...] what can I do to make it work good?

Define "work good" and tell what it is doing wrong compared to what you are expecting. Otherwise, what you mention is as useful as telling "A doesn't work as I want. How do I fix it?". I believe your first question would be something like "How do you want to to work?", right?

> Also how do I define the path for the languages [...]?

You don't. Just add the correspondent path to the relevant language file in the HTML in a script tag.

> Also how do I define the path for the [...] emotions folder?

http://www.sceditor.com/documentation/options/#emoticonsRoot

**Comment by master412160** [comment]
Hey that sounds awesome! 

I don't know what your current process is with hte integradtion but what needs to be fixed:

Is that SCEditor detects when its being hooked into phpbb and when that happens, some things of how it works, change. Like say
* bulletlist
* orderedlist
* quote
* font-size
* font
* center
* left
* right
* etc...

I guess I could manually create a bbcode for stuff like left, right, center. But the more complex like font-size, font-family might prove harder.


Because phpbb doesn't detect them when they are posted in that format, eg:
[font=Impact]fgdfgdf[/font]

But font-size it does detect but the display is off, font-size 7 is hardly visible on display. XD


The most important feature would be the quoting. Its key to the interaction. 

Second priority and more optional at this stage; is that somehow we can get the build in phpbb notification system to work when you get quoted with the quote bbcode.

Also the ability to multi quote when being in reply mode is key.


**In short:** get quote to work and multi quote (most important)

**Comment by brunoais** [comment]
**Offtopic warning; contains stuff specific to phpBB**

> Is that SCEditor detects when its being hooked into phpbb and when that happens, some things of how it works

That does not happen and I hope it never happens. phpBB is integrating SCE and not the other way around. SCE should **not** have any code specific to any existing and/or future software.

If the default BBCode of the BBCode plugin does not match phpBB's BBCode, then use a custom tag with the same name.

> The most important feature would be the quoting. Its key to the interaction. 

What's wrong with the quoting?

> Second priority and more optional at this stage; is that somehow we can get the build in phpbb notification system to work when you get quoted with the quote bbcode.

Unrelated to SCE.

> Also the ability to multi quote when being in reply mode is key.

How is that related to SCE?

**Comment by master412160** [comment]
> If the default BBCode of the BBCode plugin does not match phpBB's BBCode, then use a custom tag with the same name.

Not all bbcodes by this plugin can be remade with the custom bbcode editor of the phpbb admin panel. And some bbcodes like font-size are the same but display not the same. 

> What's wrong with the quoting?

Having a time stamp feature is all I can think of. 

> How is that related to SCE?

Well I guess it isn't but you wanted to help me or am I mistaken?

**Comment by brunoais** [comment]
> Not all bbcodes by this plugin can be remade with the custom bbcode editor of the phpbb admin panel. 

I meant an SCE custom BBCode, not phpBB custom BBCode. In my integration code, for example, all BBCode tags and all button commands are custom tags/commands from SCE's POV.

> Having a time stamp feature is all I can think of.

Do you mean SCE quote tag has a timestamp and it shouldn't?

> Well I guess it isn't but you wanted to help me or am I mistaken?

I told you I'd help with SCE and phpBB<->SCE integration. Multi-quote is exclusive to the host program and agnostic to the editor used.

**Comment by master412160** [comment]
Could you share your code then?

**Comment by brunoais** [comment]
https://github.com/phpbb/phpbb/compare/master...brunoais:feature/SCEditor

**Comment by CarpCharacin** [comment]
The pull request that @brunoais created was closed and now the goal is to have SCEditor in phpbb 3.3.  I think that a more basic integration is the way to go.  I think that the integration should be as simple as possible.  I am not sure what templates to define the SCE  JS & CSS in or the replace all text areas and BBcode code.  Can anyone help me out?  I would really like to see SCE integrated in phpbb 3.3.  I don't think that we should try to integrate SCEditor into the quick reply box in phpbb.  I think that we should try to just do a basic integration in the main text areas such as the new topic editor and the topic reply editor.  Let me know what you think about the quick reply.  I think that we should remove the emoticon button on the editor bar because phpbb has the emoticons located next to the editor.

**Comment by chilauhe** [comment]
I've done it years ago, my solution is to control the toolbar buttons displayed on textbox by a short JavaScript expression. 
It's intergraded via modifying part of php code of phpbb and templates.
It works well, but now I'm not able to find the code again due to a loss of my data.





-------- Original message --------
Sender: James Stewart <notifications@github.com>
Time: Tue 9/20 13:49
To: samclarke/SCEditor <SCEditor@noreply.github.com>
 Subject: Re: [samclarke/SCEditor] How would I make this plugin work better with a forum system like phpbb? (#437)

**Comment by CarpCharacin** [comment]
@chilauhe Let me know if you find the code.
